### PR TITLE
Add check escrow isPaid function

### PR DESCRIPTION
**Pull request by jin-tron**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior?** (You can also refer to a JIRA ticket here)
https://tronnetwork.atlassian.net/browse/BTFS-1169
check escrow payment for a contract

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)
No
* **Description of changes**
Payment work flow query escrow GRPCc interface to see if a contract has been paid in.

---

* **Please check if the PR fulfills these requirements**
- [*] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [*] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [x] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [x] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [x] Code changes have run through `go fmt`
- [x] Code changes have run through `go mod tidy`
- [ ] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

**Comment by jin-tron** [comment]
This PR is for call payin, not for the whole pay flow and guard interaction. While ledger is not working till now, please show me where's other unit test and e2e setup.

File: core/commands/storage.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ import (
 	ds "github.com/ipfs/go-datastore"
 	"github.com/libp2p/go-libp2p-core/peer"
 	ma "github.com/multiformats/go-multiaddr"
+	"go.uber.org/zap"
```
Comment (taiyangc) [inline core/commands/storage.go]: use log, not zap in btfs

File: core/commands/storage.go [hunk 2]
```diff
@@ -807,20 +808,60 @@ func SignContractAndCheckPayment(shardInfo *storage.Shards, shardHash string, ss
...
+		for {
+			select {
+			case t := <-ticker.C:
+				log.Debug("Tick at", zap.Any("time", t.UTC()))
+				paid, err = escrow.IsPaidin(configuration, contractID)
+				if err != nil {
+					log.Error("call IsPaid failed", zap.Error(err))
+				}
+				if paid {
+					paidIn <- true
```
Comment (taiyangc) [inline core/commands/storage.go]: need to return `paid` even if false.

Comment (taiyangc) [reply core/commands/storage.go]: I mean, once paid, this routine should end.

File: core/commands/storage.go (hunk 1 above)
Comment (jin-tron) [reply core/commands/storage.go]: fixed

File: core/commands/storage.go (hunk 2 above)
Comment (jin-tron) [reply core/commands/storage.go]: fixed

File: core/escrow/escrow.go [hunk 3]
```diff
@@ -233,3 +233,46 @@ func UnmarshalEscrowContract(marshaledBody []byte) (*escrowpb.SignedEscrowContra
...
+		Data:      contractID,
+		Signature: sig,
+	}, nil
+}
+
+func IsPaidin(configuration *config.Config, contractID *escrowpb.SignedContractID) (bool, error) {
+	var signedPayinRes *escrowpb.SignedPayinStatus
+	err := grpc.EscrowClient(configuration.Services.EscrowDomain).WithContext(context.Background(),
+		func(ctx context.Context, client escrowpb.EscrowServiceClient) error {
+			res, err := client.IsPaid(context.Background(), contractID)
```
Comment (taiyangc) [inline core/escrow/escrow.go]: use ctx

File: core/escrow/escrow.go [hunk 4]
```diff
@@ -233,3 +233,46 @@ func UnmarshalEscrowContract(marshaledBody []byte) (*escrowpb.SignedEscrowContra
...
+
+	return &escrowpb.SignedContractID{
+		Data:      contractID,
+		Signature: sig,
+	}, nil
+}
+
+func IsPaidin(configuration *config.Config, contractID *escrowpb.SignedContractID) (bool, error) {
+	var signedPayinRes *escrowpb.SignedPayinStatus
+	err := grpc.EscrowClient(configuration.Services.EscrowDomain).WithContext(context.Background(),
```
Comment (taiyangc) [inline core/escrow/escrow.go]: should use ctx from caller

File: core/commands/storage.go [hunk 5]
```diff
@@ -845,16 +845,58 @@ func signContractAndCheckPayment(shardInfo *storage.Shards, ssID string, n *core
...
+		return
+	}
+	signedContractID, err := escrow.SignContractID(escrowContract.ContractId, n.PrivateKey)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	paidIn := make(chan bool)
+	checkPaymentFromClient(req.Context, paidIn, signedContractID, cfg)
```
[the remark this answers is not in the document]
Comment (jin-tron) [reply core/commands/storage.go]: fixed

File: core/escrow/escrow.go (hunk 3 above)
Comment (jin-tron) [reply core/escrow/escrow.go]: fixed

File: core/escrow/escrow.go (hunk 4 above)
Comment (jin-tron) [reply core/escrow/escrow.go]: fixed

**Pull request merged**